### DISCUSSION TITLE
Is there any way to override invalidJson in Laravel 11

### DISCUSSION THREAD

**Discussion by lvqingan** [laravel/framework#50641] [Q&A]
In the previous version, I was able to override invalidJson at App\Exceptions\Handler. However, in Laravel 11, this Handler has been removed. How can I achieve it using withExceptions in `bootstrap/app.php`?

```php
class Handler extends ExceptionHandler
{
    // ...

    protected function invalidJson($request, ValidationException $exception)
    {
        return response()->json([
            'message' => $exception->getMessage(),
            'error'   => SystemErrorCode::FormValidateFailed->value,
            'data'    => $exception->errors()
        ], 200);
    }
}
```

**Comment by lvqingan** [comment]
Found a solution, I can rebind ExceptionHandler contract to my own Handler. But the code doesn't look very elegant.

```php
$app = Application::configure(basePath: dirname(__DIR__))
    // ...
   ->create();

$app->singleton(
    \Illuminate\Contracts\Debug\ExceptionHandler::class,
    \App\Exceptions\Handler::class
);

$using = function (Exceptions $exceptions) {
    // ...
};

$app->afterResolving(
    \App\Exceptions\Handler::class,
    fn ($handler) => $using(new Exceptions($handler)),
);

return $app;
```

**Comment by lvqingan** [comment]
Got a much better answer from laracasts

```php
return Application::configure(basePath: dirname(__DIR__))
    ->withExceptions(function (Exceptions $exceptions) {
        $exceptions->render(function (ValidationException $e, Request $request) {
            if ($request->expectsJson()) {
                return response()->json([
                    'message' => $e->getMessage(),
                    'error'   => SystemErrorCode::FormValidateFailed->value,
                    'data'    => $e->errors()
                ], 200);
            }

            return null;
        });
    })->create();
```

**Reply by ramtinq** [reply]
> Got a much better answer from laracasts
> 
> ```php
> return Application::configure(basePath: dirname(__DIR__))
>     ->withExceptions(function (Exceptions $exceptions) {
>         $exceptions->render(function (ValidationException $e, Request $request) {
>             if ($request->expectsJson()) {
>                 return response()->json([
>                     'message' => $e->getMessage(),
>                     'error'   => SystemErrorCode::FormValidateFailed->value,
>                     'data'    => $e->errors()
>                 ], 200);
>             }
> 
>             return null;
>         });
>     })->create();
> ```

Well for some reason this doesn't work for custom handling of AuthenticationException. The page still redirects to the redirectTo:

```php
    ->withExceptions(function (Exceptions $exceptions) {
        $exceptions->render(function (AuthenticationException $exception, Request $request) {
            return 'tessst'; /// or do whatever redirecting, nothing works.



        });
    })
```
